webi-docker.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/sh
+#shellcheck disable=SC2046
+
+VERSION_OF_UBUNTU_CONTAINER="22.04"
+WEBI_CONTAINER_FOLDER="${1:-""}"
+
+timestamp() {
+    echo "[+] $(date +'%F %T') [INFO] $*"
+    echo
+}
+
+timestamp_ending() {
+    echo "[+] $(date +'%F %T') [INFO] $*"
+}
+
+err() {
+    echo "[-] $(date +'%F %T') [ERROR] $*" >&2
+    echo
+}
+
+command_start() {
+    timestamp "Command $* has been started."
+    if ! "$@"; then
+        err "Command $* went wrong."
+        exit
+    fi
+    timestamp "Command $* has been ended."
+}
+
+remove_old_docker_containers() {
+    container="$(docker ps --all | grep -i Webi | awk '{print $1}')"
+    if [ -n "$container" ]; then
+        docker stop "$container" 2>/dev/null
+        docker rm "$container" 2>/dev/null
+    fi
+}
+
+pull_and_run_ubuntu_container() {
+    docker pull jrei/systemd-ubuntu:$VERSION_OF_UBUNTU_CONTAINER
+    docker run \
+        -d \
+        $([ -z "$WEBI_CONTAINER_FOLDER" ] || echo " -v ""$WEBI_CONTAINER_FOLDER"":/opt") \
```

Comment:
You can use `${WEBI_CONTAINER_FOLDER:-.}` to default to `.`.

- `:-<value>` means set to `<value>` if not set, or if empty

Reply:
When I changed it to:
```diff
diff --git a/webi-docker.sh b/webi-docker.sh
index e56ef07..d7ba735 100755
--- a/webi-docker.sh
+++ b/webi-docker.sh
@@ -41,7 +41,7 @@ pull_and_run_ubuntu_container() {
     docker pull jrei/systemd-ubuntu:$VERSION_OF_UBUNTU_CONTAINER
     docker run \
         -d \
-        $([ -z "$WEBI_CONTAINER_FOLDER" ] || echo " -v ""$WEBI_CONTAINER_FOLDER"":/opt") \
+        -v "${WEBI_CONTAINER_FOLDER:-.}":/opt \
```
That script is not working.

```
yorune@MacBook ~/git/webi-installers webi-add-docker $ ./webi-docker.sh
[+] 2022-08-29 23:52:31 [INFO] Command remove_old_docker_containers has been started.

ea9eb78d0dc6
ea9eb78d0dc6
[+] 2022-08-29 23:52:46 [INFO] Command remove_old_docker_containers has been ended.

[+] 2022-08-29 23:52:46 [INFO] Command pull_and_run_ubuntu_container has been started.

22.04: Pulling from jrei/systemd-ubuntu
Digest: sha256:b99be1d2288bf7938e0bff3c142c8f8dc9c1cd0e6149f348217497caaf231b78
Status: Image is up to date for jrei/systemd-ubuntu:22.04
docker.io/jrei/systemd-ubuntu:22.04
docker: Error response from daemon: create .: volume name is too short, names should be at least two alphanumeric characters.
See 'docker run --help'.
[-] 2022-08-29 23:52:49 [ERROR] Command pull_and_run_ubuntu_container went wrong.

Execution time: 0h:00m:18s sec
```


I applied that echo due to the `-v ...` needs a docker command. Maybe do you know another idea to apply that feature?

Reply:
I only use Docker when I have to, but I thought that the mapping was `<localdir>:<dockerdir>`?

Usually the docker user is `app` and the home folder is `/app` and so webi would create `/app/.local/bin/` and `/app/.local/opt/` and run the same as on desktops.

`npm` should NOT be preinstalled. If there's anything that errors out because npm isn't there, that's a bug that needs to be fixed.

Rather than `jrei/systemd-ubuntu`, the standard `Alpine` should work now.

Maybe we can get on a voice call and you can educate me on how you think this ought to work and if I can provide some feedback and maybe adjust some things in webi?